File: the-nest-frontend/types/post.ts
export interface Post {
  id: string
  user_id: string
  content: string
  parent_post_id: string | null
  is_reply: boolean
  like_count: number
  reply_count: number
  repost_count: number
  edit_count: number
  media_url?: string | null
  media_type?: string | null
  created_at: string
  updated_at: string
  reposted_by?: string
  user?: {
    username: string
    display_name: string
    avatar_url: string | null
    role: string
  }
}

File: the-nest-backend/app/schemas/post.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


class PostCreate(BaseModel):
    content: str = Field(..., max_length=280)
    parent_post_id: Optional[UUID] = None
    media_url: Optional[str] = None
    media_type: Optional[str] = None


class PostUpdate(BaseModel):
    content: str = Field(..., max_length=280)


class PostAuthor(BaseModel):
    username: str
    display_name: str
    avatar_url: Optional[str]
    role: str

    model_config = {"from_attributes": True}


class PostResponse(BaseModel):
    id: UUID
    user_id: UUID
    content: str
    parent_post_id: Optional[UUID] = None
    is_reply: bool
    like_count: int
    reply_count: int
    repost_count: int
    edit_count: int
    media_url: Optional[str] = None
    media_type: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PostWithAuthor(PostResponse):
    user: Optional[PostAuthor] = None
    reposted_by: Optional[str] = None
